stl/inc/vector

```diff
@@ -2882,6 +2882,62 @@ namespace pmr {
     using vector = _STD vector<_Ty, polymorphic_allocator<_Ty>>;
 } // namespace pmr
 #endif // _HAS_CXX17
+
+#if _HAS_IF_CONSTEXPR
+// VARIABLE TEMPLATE _Is_vb_iterator
+template <class _Alloc>
+_INLINE_VAR constexpr bool _Is_vb_iterator<_Vb_const_iterator<_Alloc>> = true;
+
+template <class _Alloc>
+_INLINE_VAR constexpr bool _Is_vb_iterator<_Vb_iterator<_Alloc>> = true;
+
+template <class _FwdIt, class _Ty>
+_CONSTEXPR20 void _Fill_vbool(_FwdIt _First, _FwdIt _Last, const _Ty& _Val) {
+    // Set [_First, _Last) to _Val
+    if (_First == _Last) {
+        return;
+    }
+
+    _Vbase* _UFirst            = const_cast<_Vbase*>(_First._Myptr);
+    const _Vbase* const _ULast = _Last._Myptr;
+
+    const auto _FirstSourceMask = static_cast<_Vbase>(-1) << _First._Myoff;
+    const auto _LastSourceMask  = static_cast<_Vbase>(-1) >> (_VBITS - _Last._Myoff);
+    const auto _FirstDestMask   = static_cast<_Vbase>(~_FirstSourceMask);
+    const auto _LastDestMask    = static_cast<_Vbase>(~_LastSourceMask);
+    const auto _FillVal         = static_cast<_Vbase>(_Val ? -1 : 0);
+
+    if (_UFirst == _ULast) {

```

Comment:
This branch also seems unnecessary?

Reply:
It is not. Otherwise we would clobber memory

Note that the algorithm is essentially 3 steps.

1. Fill first block from `_First._Myoff` 

2. Fill blocks between `_First._Myptr` and `_Last._Myptr`

3. If `_Last._Myoff != 0` then fill the last block until `_Last._Myoff != 0`

Now it we would not do this check here we would write beyond `_Last._Myoff` in the first step and also clobber everything before `_First._Myoff`

Reply:
I woult like to note that the `copy` version has 3 different special cases as that reasoning applies to both `_Source` and `_Dest`

Reply:
Right, I'm suggesting a form that does:

1. Fill the leading bits of the first block. (if and only if the entire first block isn't being overwritten)
2. Fill blocks in between, if any
3. Fill trailing bits of the last block. (if and only if the entire last block isn't being overwritten)

which should eliminate both this branch and the special case above. And we need branches for those if and only ifs anyway to avoid performing the full shift.

Reply:
Hmmm I suppose my proposal would still need the special case because when the first and last block are the same the first step would clobber trailing bits. It still seems like once this is fixed to avoid the full shift there will be opportunities to merge the branch necessary for that with these other checks....

Reply:
The problem is that if `_First` and `_Last` point to the same block you need to *and* the masks and not *or*, which is what would happen if you do it in the two steps

Reply:
BTW I might have written it differently buit your proposal is the same as my proposal. 

Also without this if `_First `and `_Last `point ot the same block the sequence would -effectively- shorten to

1. Fill preserving the leading bits of the block. (if and only if the entire first block isn't being overwritten)
2. Fill preserving trailing bits of the block. (if and only if the entire last block isn't being overwritten)

As you can see 1. will clobber `[_Last._Myoff, 32)` and 2. will clobber` [0, _First._Myoff)`

Reply:
Yes, you're right. I think I've reduced the difference I want to see to just avoiding the forbidden full shift, see my comment below.